packages/core/src/StateNode.ts

```diff
@@ -353,7 +354,11 @@ export class StateNode<
   public next(
     state: State<TContext, TEvent>,
     _event: SCXML.Event<TEvent>
-  ): StateTransition<TContext, TEvent> | undefined {
+  ): Transitions<TContext, TEvent> | undefined {
+    if (this.type === 'history' && state.historyValue[this.id]) {
```

Comment:
this is a little bit confusing to me - why do we have to have this check here? If we have stored historyValue for a particular id then we should enter those states, not return an empty array.

Seems like this is handled upstream in the `resolveHistory` already and this condition can be removed, or do I miss something?